packages/core/bootstrap/src/lib/cache-warmer/epics.ts

```diff
@@ -177,17 +181,20 @@ export const warmupSubscriber: Epic<AnyAction, AnyAction, any, EpicDependencies>
       const pollInterval = interval - offset
       return timer(pollInterval, pollInterval).pipe(
         mapTo(warmupRequested({ key })),
+        takeUntil(action$.pipe(filter(warmupShutdown.match))),
         // unsubscribe our warmup algo when a matching unsubscribe comes in
         takeUntil(
           action$.pipe(
             filter(warmupUnsubscribed.match || warmupStopped.match),
-            filter((a) => a.payload.key === key),
+            filter(
+              (a: PayloadAction<WarmupUnsubscribedPayload | WarmupStoppedPayload>) =>
+                ('key' in a.payload && a.payload.key === key) ||
+                ('keys' in a.payload && a.payload.keys.includes(key)),
+            ),
```

Comment:
It seemed like this was not really working previously, so I included a fix here. Should I keep or remove?

Reply:
IMO this looks good. There may be more condensed alternatives but I like the readability of it.